### PR TITLE
92430 stop loading subscribers from KV

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.DirectorService/Program.cs [hunk 1]
```diff
@@ -59,10 +58,6 @@ private static async Task Main()
                     .As<ISubscriberConfigurationLoader>()
                     .SingleInstance();
 
-                builder.RegisterType<ConfigurationMerger>()
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/Program.cs:62]: There's a SubscribersKeyVaultProvider on line 58 which needs to be removed as well, cannot highlight it from here

**Review by tabman83 — CHANGES_REQUESTED** [review]

File: src/CaptainHook.DirectorService/Program.cs (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.DirectorService/Program.cs:62]: Also both ISubscribersKeyVaultProvider.cs and SubscribersKeyVaultProvider.cs needs to be removed

Comment (tabman83) [reply src/CaptainHook.DirectorService/Program.cs:62]: And its test files if any

**Review by tabman83 — CHANGES_REQUESTED** [review]

File: src/CaptainHook.DirectorService/Program.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Program.cs:62]: Good idea! Removed, as well as KeyVaultConfigurationLoader

**Review by tabman83 — APPROVED** [review]
Please hold on to the merge for now

File: src/CaptainHook.DirectorService/Infrastructure/SubscriberConfigurationLoader.cs [hunk 2]
```diff
@@ -11,32 +16,50 @@ namespace CaptainHook.DirectorService.Infrastructure
...
+            }
+
+            try
+            {
+                await Task.WhenAll(tasks);
+            }
+            catch (Exception ex)
             {
-                return subscribersFromKV.Error;
+                return new MappingError("Cannot map Cosmos DB entries", new ExceptionFailure(ex));
```
Comment (chomick) [inline src/CaptainHook.DirectorService/Infrastructure/SubscriberConfigurationLoader.cs:52]: Should you check here which tasks had exceptions and log those?

Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Infrastructure/SubscriberConfigurationLoader.cs]: I've changed that part, so now we know the ID of failed subscriber entity.
Logging is done by DirectorService.

The question is - do we really should fail DirectorService if one subscriber configuration is broken?

File: src/CaptainHook.DirectorService/Infrastructure/Interfaces/LoadingSubscribersResult.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using CaptainHook.Common.Configuration;
+using CaptainHook.Domain.Results;
+
+namespace CaptainHook.DirectorService.Infrastructure.Interfaces
+{
+    public class LoadingSubscribersResult
+    {
+        public IEnumerable<ErrorBase> Errors { get; }
+        public IEnumerable<SubscriberConfiguration> Subscribers { get; }
+        public bool HaveErrors => Errors.Any();
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/Infrastructure/Interfaces/LoadingSubscribersResult.cs:13]: Not meaning to be pedantic, I'd say `HasErrors`

Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Infrastructure/Interfaces/LoadingSubscribersResult.cs:13]: Sure, will fix

File: src/CaptainHook.DirectorService/Infrastructure/Interfaces/LoadingSubscribersResult.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CaptainHook.Common.Configuration;
+using CaptainHook.Domain.Results;
+
+namespace CaptainHook.DirectorService.Infrastructure.Interfaces
+{
+    public class LoadingSubscribersResult
```
Comment (chomick) [inline src/CaptainHook.DirectorService/Infrastructure/Interfaces/LoadingSubscribersResult.cs:9]: I didn't think we have to tell the caller what worked what did not? I though we could simply log the individual errors and return correct subscribers?

**Review by chomick — COMMENTED** [review]
Before we merge this, is there any way to verify (on any env) that there is no longer ANY keyvault config loaded and all of them are already served from DB only?
Do we have some indicator on GET operation on where the config came from?

**Pull request merged**